packages/server/src/course/course.controller.ts

```diff
@@ -893,4 +896,84 @@ export class CourseController {
       });
     return;
   }
+
+  // UPDATE course_settings_model SET selectedFeature = false WHERE courseId = selectedCourseId;
+  // will also create a new course settings record if it doesn't exist for the course
+  @Patch(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async enableDisableFeature(
+    @Param('id') courseId: number,
+    @Body() body: CourseSettingsRequestBody,
+  ): Promise<void> {
+    // fetch existing course settings
+    let courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId: courseId },
+    });
+
+    // if no course settings exist yet, create new course settings for the course
+    if (!courseSettings) {
+      // first make sure the course exists in course table (this check might not be needed as the guards already make sure the user is in the course (therefore the course must exist), but this is a rare function to be called so the small performance hit is acceptable for later safety)
+      const course = await CourseModel.findOne({
+        where: { id: courseId },
+      });
+      if (!course) {
+        throw new NotFoundException(
+          'Error while creating course settings: Course not found',
+        );
+      }
+      courseSettings = new CourseSettingsModel(); // all features are enabled by default, adjust in CourseSettingsModel as needed
+      courseSettings.courseId = courseId;
+    }
+
+    // then, toggle the requested feature
+    try {
+      courseSettings[body.feature] = body.value;
+    } catch (err) {
+      throw new BadRequestException('Invalid feature');
+    }
+
+    try {
+      await courseSettings.save();
+    } catch (err) {
+      throw new HttpException(
+        'Error while saving course settings',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  // SELECT * FROM course_settings_model WHERE courseId = selectedCourseId;
+  // if no settings for the courseid, return all true
+  @Get(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR, Role.STUDENT, Role.TA)
+  async getFeatures(
+    @Param('id') courseId: number,
+  ): Promise<CourseSettingsResponse> {
+    const courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId },
+    });
+
+    // if no settings found, do default
+    if (!courseSettings) {
+      return new CourseSettingsResponse({
+        courseId: courseId,
+        chatBotEnabled: true,
+        asyncQueueEnabled: true,
+        adsEnabled: true,
+        queueEnabled: true,
+        settingsFound: false,
+      });
+    }
+
+    return new CourseSettingsResponse({
+      courseId: courseId,
+      chatBotEnabled: courseSettings.chatBotEnabled,
```

Comment:
Shouldn't we return something like this?

```
return res.status(... status code here).send({courseId, ...courseSettings, settingsFound });
```

This will ensure that in future if we add more feature we don't need to update it in multiple places

Reply:
will change, thanks